p2p/src/node.rs

```diff
@@ -694,109 +713,115 @@ where
+                                || protocol == &kad::PROTOCOL_NAME
+                            {
+                                for addr in &info.listen_addrs {
+                                    if let Some(kad) = self.swarm.behaviour_mut().kad.as_mut() {
+                                        kad.add_address(&peer_id, addr.clone());
+                                        kad_address_added = true;
+                                    }
                                 }
-                            }
-                        } else if protocol == &StreamProtocol::new("/libp2p/autonat/1.0.0") {
-                            // TODO: expose protocol name on `libp2p::autonat`.
-                            // TODO: should we remove them at some point?
-                            for addr in &info.listen_addrs {
-                                if let Some(autonat) = self.swarm.behaviour_mut().autonat.as_mut() {
-                                    autonat.add_server(peer_id, Some(addr.clone()));
+                            } else if protocol == &StreamProtocol::new("/libp2p/autonat/1.0.0") {
+                                // TODO: expose protocol name on `libp2p::autonat`.
+                                // TODO: should we remove them at some point?
+                                for addr in &info.listen_addrs {
+                                    if let Some(autonat) =
+                                        self.swarm.behaviour_mut().autonat.as_mut()
+                                    {
+                                        autonat.add_server(peer_id, Some(addr.clone()));
+                                    }
                                 }
                             }
                         }
-                    }
-                    if let Some(bitswap) = self.swarm.behaviour().bitswap.as_ref() {
-                        bitswap.on_identify(&peer_id, &info.protocols);
-                    }
+                        if let Some(bitswap) = self.swarm.behaviour().bitswap.as_ref() {
+                            bitswap.on_identify(&peer_id, &info.protocols);
+                        }
 
-                    self.swarm
-                        .behaviour_mut()
-                        .peer_manager
-                        .inject_identify_info(peer_id, info.clone());
+                        self.swarm
+                            .behaviour_mut()
+                            .peer_manager
+                            .inject_identify_info(peer_id, info.clone());
 
-                    if let Some(channels) = self.lookup_queries.remove(&peer_id) {
-                        for chan in channels {
-                            chan.send(Ok(info.clone())).ok();
+                        if let Some(channels) = self.lookup_queries.remove(&peer_id) {
+                            for chan in channels {
+                                chan.send(Ok(info.clone())).ok();
```

Comment:
Do we care to drop closed channels? (I know you didn't change this), just curious

Reply:
nvm.. i didn't realize we removed it from the map already.. so it's good